storage/filesystem_storage.go

```diff
@@ -0,0 +1,167 @@
+package storage
+
+import (
+	"context"
+	"github.com/gazebo-web/gz-go/v7"
+	"github.com/pkg/errors"
+	"io"
+	"os"
+)
+
+// fileSys contains the implementation of a storage manager for resources using the host filesystem.
+// It can be used with AWS EFS storage in EC2 instances.
+type fileSys struct {
+	basePath string
+}
+
+// UploadZip uploads the given file as the zip file of the given resource.
```

Comment:
I find it hard to understand what this function does from this documentation alone. 

From the implementation I can see that it expects to receive a zip file to upload., but a question that came to me from reading it was: Is `file` supposed to be an already zipped file that you want to upload or is it supposed to be a file descriptor of some path (which can be either a file or directory) that is to be zipped by this function?

Reply:
Resources can have a compressed representation of the resource itself, it contains all the files from the said resource. This function uploads that zip file.